client/lib/catch-js-errors/index.js

```diff
@@ -120,12 +119,12 @@ export default class ErrorLogger {
 	}
 
 	sendToApi( error ) {
-		const xhr = new XMLHttpRequest();
-		xhr.open( 'POST', 'https://public-api.wordpress.com/rest/v1.1/js-error?http_envelope=1', true );
-		xhr.setRequestHeader( 'Content-type', 'application/x-www-form-urlencoded' );
-		let params =
-			'client_id=39911&client_secret=cOaYKdrkgXz8xY7aysv4fU6wL6sK5J8a6ojReEIAPwggsznj4Cb6mW0nffTxtYT8&error=';
-		params += encodeURIComponent( JSON.stringify( error ) );
-		xhr.send( params );
+		const body = new window.FormData();
+		body.append( 'error', JSON.stringify( error ) );
+
+		window.fetch( 'https://public-api.wordpress.com/rest/v1.1/js-error?http_envelope=1', {
```

Comment:
If we don't look at the response, then the `http_envelope=1` param is not needed: it determines the shape of the response.

Also, `window.fetch()` may fail and return rejected promise. Would be nice to catch it formally to avoid unhandled rejection events.

Reply:
> If we don't look at the response, then the `http_envelope=1` param is not needed: it determines the shape of the response.

Good call. Removed in https://github.com/Automattic/wp-calypso/pull/47003/commits/9bdafa2ad1d037f0b0b0c7961cf2e3f94945c99b

> Also, `window.fetch()` may fail and return rejected promise. Would be nice to catch it formally to avoid unhandled rejection events.

An error occurs when attempting to log an error? 🤯

Anyway, done in https://github.com/Automattic/wp-calypso/pull/47003/commits/b92ceb6b0791274299e3f87529e48ed3ff5727ea.

Reply:
> An error occurs when attempting to log an error?

Yes, and it's not ruled out that the JS catch handler will try to report it recursively 🙂

Reply:
Sure. I think logging the error in the console is good enough at this point, but let me know if you have other suggestions.